### ISSUE TITLE
Angular 14 support?

### ISSUE THREAD

**Issue by SamuelMarks** [angular-schule/angular-cli-ghpages#148]
There's a new version out!

**Comment by JohannesHoppe** [comment]
Thanks for the reminder, I will pump the versions. But there should be no breaking changes for us. `--force` should do the trick until then.

**Comment by Mijago** [comment]
I just wanted to note that with Angular 14 I get: 
> ng add angular-cli-ghpages
> [...]
> Package "angular-cli-ghpages" was found but does not support schematics.

**Comment by fmalcher** [comment]
@Mijago That's because the Angular CLI automatically detects the latest compatible version – and in this case it cannot find one.
Since a few things have changed with v14, there was a tiny bit more to do. PR is ready: https://github.com/angular-schule/angular-cli-ghpages/pull/149

**Comment by JohannesHoppe** [comment]
Unfortunately this is blocked by: https://github.com/angular/angular-cli/issues/23397
**tl;dr:** Arguments like `noBuild`, `noSilent`, `noDotFiles` no longer work

Let's wait for a statement of the Angular Team (if this is a bug/breaking change or if we have to adjust)

**Comment by fmalcher** [comment]
As long as we cannot release a proper version for Angular 14, you have two options:
* Use the standalone version (see README_standalone)
* Install the package manually: `npm i angular-cli-ghpages --force` and then add the builder config in `angular.json` manually, under `projects.YOURPROJECT.architect`, next to the `build` config:

```
"deploy": {
  "builder": "angular-cli-ghpages:deploy"
}
```

Please note that the documented options `noBuild`, `noSilent` and `noDotFiles` don't work with Angular 14 anymore.

**Comment by JohannesHoppe** [comment]
I fix was created:
https://github.com/angular/angular-cli/pull/23405 and will be available in the next CLI version.

Users will see the following message:

>  Warning: '${name}' option has been declared with a 'no' prefix in the schema. Please file an issue with the author of this package.

A) Do we want to rely on the fix (users will see the message) and push a fast version out?
B) Or do we want to refactor all arguments with a big breaking change on our side?

**Comment by SamuelMarks** [comment]
It also looks like the default project is being removed and `ngh` doesn't know how to proceed:
```
No Angular project selected and no default project in the workspace
```

…maybe if there's only one non-test project in the workspace you auto-treat that as the default project?

**Comment by fmalcher** [comment]
angular-eslint has the same problem with the `no` negation: https://github.com/angular-eslint/angular-eslint/issues/1069

**Comment by JohannesHoppe** [comment]
Some great progress here, the CLI will get the old behavior back and there will be no warning anymore:

> Therefore, in order to reduce the community changes we remove the deprecation warning for such options and change the interim solution to permanent one.

https://github.com/angular/angular-cli/pull/23536

**Comment by JohannesHoppe** [comment]
All tests are green now. ✅
I just pushed [v1.0.1](https://github.com/angular-schule/angular-cli-ghpages/releases/tag/v1.0.1) to NPM! Happy coding! 😄

**Comment by SamuelMarks** [comment]
@JohannesHoppe Thanks, but getting errors:
See https://github.com/SamuelMarks/SamuelMarks-www/runs/7521268473

```
Cannot find module 'node_modules/angular-cli-ghpages/ng-add'
Require stack:
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/src/command-builder/utilities/schematic-engine-host.js
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/src/command-builder/schematics-command-module.js
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/src/commands/add/cli.js
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/src/command-builder/command-runner.js
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/lib/cli/index.js
- /opt/hostedtoolcache/node/16.16.0/x64/lib/node_modules/@angular/cli/lib/init.js
```

After running `ng add` @ https://github.com/SamuelMarks/SamuelMarks-www/actions/runs/2739862339/workflow#L25

**Comment by fmalcher** [comment]
@SamuelMarks I can confirm! We're working on it. Sorry for the inconvenience …

**Comment by JohannesHoppe** [comment]
@SamuelMarks I blame npm for this! The command line arguments of NPM 8 have been changed! 🙈 

It's now fixed, thanks for the fast report!
Please try out v1.0.3

**Comment by SamuelMarks** [comment]
Great that worked! - Thanks
